pkg/operator/ceph/config/datapath.go

```diff
@@ -47,35 +53,42 @@ type DataPathMap struct {
 // which may include data from other daemons.
 func NewStatefulDaemonDataPathMap(
 	dataDirHostPath, daemonDataDirHostRelativePath string,
-	daemonType DaemonType, daemonID string,
+	daemonType DaemonType, daemonID, namespace string,
 ) *DataPathMap {
 	return &DataPathMap{
 		PersistData:      true,
 		HostDataDir:      path.Join(dataDirHostPath, daemonDataDirHostRelativePath),
 		NoData:           false,
 		ContainerDataDir: cephDataDir(daemonType, daemonID),
+		HostLogDir:       path.Join(dataDirHostPath, namespace, "log"),
+		ContainerLogDir:  VarLogCephDir,
 	}
 }
 
 // NewStatelessDaemonDataPathMap returns a new DataPathMap for a daemon which does not persist data
 // to the host (mgrs, mdses, rgws)
 func NewStatelessDaemonDataPathMap(
-	daemonType DaemonType, daemonID string,
+	daemonType DaemonType, daemonID, namespace, dataDirHostPath string,
 ) *DataPathMap {
 	return &DataPathMap{
 		PersistData:      false,
 		HostDataDir:      "",
 		NoData:           false,
 		ContainerDataDir: cephDataDir(daemonType, daemonID),
+		HostLogDir:       path.Join(dataDirHostPath, namespace, "log"),
+		ContainerLogDir:  VarLogCephDir,
 	}
 }
 
-func NewDatalessDaemonDataPathMap() *DataPathMap {
+// NewDatalessDaemonDataPathMap returns a new DataPathMap for a daemon which does not utilize a data dir in the container as the mon, mgr, osd, mds, and rgw daemons do
+func NewDatalessDaemonDataPathMap(daemonType DaemonType, daemonID, namespace, dataDirHostPath string) *DataPathMap {
 	return &DataPathMap{
 		PersistData:      false,
 		HostDataDir:      "",
 		NoData:           true,
-		ContainerDataDir: "",
+		ContainerDataDir: cephDataDir(daemonType, daemonID),
```

Comment:
The dataless daemon has no data dir (`NoData == true`). `daemonType` and `daemonID` are unnecessary here. I think maybe I should have mentioned in the comments that `NoData` and `ContainerDataDir` are linked; I can see how that's ambiguous.

Reply:
I just had two change suggestions this time; one of them is linked below, which I don't think is technically part of my requested changes comments: https://github.com/rook/rook/pull/2921#discussion_r274496678